hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/TimelineService.java

```diff
@@ -52,36 +53,50 @@
 
   private int serverPort;
   private Configuration conf;
+  private transient HoodieEngineContext context;
   private transient FileSystem fs;
   private transient Javalin app = null;
   private transient FileSystemViewManager fsViewsManager;
   private final int numThreads;
   private final boolean shouldCompressOutput;
   private final boolean useAsync;
+  private final int markerBatchNumThreads;
```

Comment:
its okay for timeline service to know these. its a core functionality provided by the timeline server

Reply:
These have been included in the TimelineService.Config and we use the config instance to pass all config values.